code/controllers/subsystem/atoms.dm

```diff
@@ -27,6 +27,15 @@ SUBSYSTEM_DEF(atoms)
 	initialized = INITIALIZATION_INSSATOMS
 
 /datum/controller/subsystem/atoms/Initialize(timeofday)
+	//Wait until map loading is completed
+	if (length(SSmap_generator.executing_generators) > 0)
+		to_chat(world, "<span class='boldannounce'>Waiting for [length(SSmap_generator.executing_generators)] map generators...</bold>")
+		do
+			SSmap_generator.fire()
+			sleep(0.5)
```

Comment:
Since fire() is tick checked, if it overruns then the proc will cancel. Without the sleep, byond will trigger the fire() again, which will do nothing since the tick check just instantly fails. The sleep allows the game to process to the next tick, so that the MC_TICK_CHECKs inside fire() can run again. Works similarly to the same way the main MC loop works